src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -228,11 +234,84 @@ public ISqlConnectionFactory ConnectionFactory
         public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connectionInfo) => this.OwnerToConnectionMap.TryGetValue(ownerUri, out connectionInfo);
 
         /// <summary>
-        /// Validates the given ConnectParams object. 
+        /// Refreshes the auth token of a given connection, if needed
         /// </summary>
-        /// <param name="connectionParams">The params to validate</param>
-        /// <returns>A ConnectionCompleteParams object upon validation error, 
-        /// null upon validation success</returns>
+        /// <param name="ownerUri">The URI of the connection</param>
+        /// <returns>true if a refresh is needed,
+        /// false if the uri cannot be found, refresh is not needed,
+        /// refresh token has already been requested, or if refresh fails </returns>
+
+        internal async Task<bool> TryRefreshAuthToken(string ownerUri)

```

Comment:
Let's name this `TryRequestRefreshAuthToken` instead to make it clear this doesn't immediately refresh the token